cpp/arcticdb/column_store/memory_segment_impl.cpp

```diff
@@ -513,6 +514,44 @@ void SegmentInMemoryImpl::concatenate(SegmentInMemoryImpl&& other, bool unique_c
     }
 }
 
+bool SegmentInMemoryImpl::is_row_duplicate(
```

Comment:
This should be a free function that takes a segment and a row, and should not be in ArcticDB. This duplicate detection stuff needs rewriting anyway, hashing rows one by one is incredibly inefficient